### DISCUSSION TITLE
Skip upload content to vectordb if functions/filter have files support

### DISCUSSION THREAD

**Discussion by ndrsfel** [open-webui/open-webui#5804] [Q&A] [answered]
Hello everyone,

I am using a filter with the `self.file_handler = True` attribute set. Therefore, I expect open-webui to handle files by the filter. Apparently, the default behavior is to extract the file's content and upload to the vector database as seen [here](https://github.com/open-webui/open-webui/blob/1d225dd804575af9ae5981528dfdce695f7f7040/src/lib/components/chat/MessageInput.svelte#L135) which is not required in this case:

```js
const uploadFileHandler = async (file) => {
...
		// TODO: Check if tools & functions have files support to skip this step to delegate file processing
		// Default Upload to VectorDB
		if (
			SUPPORTED_FILE_TYPE.includes(file['type']) ||
			SUPPORTED_FILE_EXTENSIONS.includes(file.name.split('.').at(-1))
		) {
			processFileItem(fileItem);
		} else {
...
```

I'd like to contribute this feature to the library. Do you know how to check if a filter with the file_handler attribute is enabled from the frontend? Help much appreciated. Thank you!

**Comment by tjbck** [comment, marked as answer]
This part has been refactored in dev, and will remain unchanged (WebUI will always try to extract the content of the file regardless of the supported status).

**Reply by ndrsfel** [reply]
This is awesome! Thank you @tjbck - Skipping RAG possible starting version 0.3.31 🎉

**Reply by tjbck** [reply]
Were you able to test it with the latest dev?

**Reply by ndrsfel** [reply]
Yes I was, tried locally and worked out of the box. Was a little bit tricky to find the toggle switch but straightforward once discovered. Do you have any plans to allow to set default behaviour globally (config) or even by model? Keep up the great work 🚀

**Reply by navneetpr** [reply]
I'm not sure which configuration helps upload only file and never start the embedding process. Any help here ?

**Reply by Podden** [reply]
Still I dont understand? So it skips but still extracts? What does it skip then? Only the adding to the System promt Part? 

I want to create a toggle to upload a pdf to the anthropic files api for remote Code Execution Tool instead of let openwebui handle it. Would be awesome as well to intercept the add file to Message Event to start uploading before User submits the request. Is this possible with the file_handler?

**Comment by xuzhougeng** [comment]
I'm not entirely sure if I understand your discussion. I haven't found any specific configuration（https://docs.openwebui.com/getting-started/advanced-topics/env-configuration#rag） to ensure that file uploads only extract content without performing additional embedding.

I noticed that after file upload, there is always a save_docs_to_vector_db operation, which always calls the embedding_function function to construct items:

See: 

https://github.com/open-webui/open-webui/blob/96c865404d36637eafadb6d2dd2365c85d452648/backend/open_webui/apps/webui/routers/files.py#L71

https://github.com/open-webui/open-webui/blob/96c865404d36637eafadb6d2dd2365c85d452648/backend/open_webui/apps/retrieval/main.py#L724-L746

but , I also found that rag can be ignore in chat 

See:

https://github.com/open-webui/open-webui/blob/96c865404d36637eafadb6d2dd2365c85d452648/backend/open_webui/apps/retrieval/utils.py#L327C1-L332C14

So, Is there a configuration that would allow save_docs_to_vector_db to only retain the text while leaving the vector field empty?"

**Reply by sir3mat** [reply]
Follow
